pkg/util/priority/priority_test.go

```diff
@@ -0,0 +1,173 @@
+		})
+	}
+}
+
+func TestGetPriorityFromPriorityClass(t *testing.T) {
+	scheme := runtime.NewScheme()
+	if err := schedulingv1.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding scheduling scheme: %v", err)
+	}
+
+	tests := []struct {
+		desc                       string
+		priorityClassList          *schedulingv1.PriorityClassList
+		priorityClassName          string
+		expectedPriorityClassName  string
+		expectedPriorityClassValue int32
+		expectedErr                string
+	}{
+		{
+			desc: "priorityClass is specified and it exists",
+			priorityClassList: &schedulingv1.PriorityClassList{
+				Items: []schedulingv1.PriorityClass{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "test"},
+						Value:      50,
+					},
+				},
+			},
+			priorityClassName:          "test",
+			expectedPriorityClassName:  "test",
+			expectedPriorityClassValue: 50,
+		},
+		{
+			desc: "priorityClass is specified and it does not exist",
+			priorityClassList: &schedulingv1.PriorityClassList{
+				Items: []schedulingv1.PriorityClass{},
+			},
+			priorityClassName: "test",
+			expectedErr:       `priorityclasses.scheduling.k8s.io "test" not found`,
+		},
+		{
+			desc: "priorityClass is unspecified and one global default exists",
+			priorityClassList: &schedulingv1.PriorityClassList{
+				Items: []schedulingv1.PriorityClass{
+					{
+						ObjectMeta:    v1.ObjectMeta{Name: "globalDefault"},
+						GlobalDefault: true,
+						Value:         40,
+					},
+				},
+			},
+			expectedPriorityClassName:  "globalDefault",
+			expectedPriorityClassValue: 40,
+		},
+		{
+			desc: "priorityClass is unspecified and multiple global defaults exist",
+			priorityClassList: &schedulingv1.PriorityClassList{
+				Items: []schedulingv1.PriorityClass{
+					{
+						ObjectMeta:    v1.ObjectMeta{Name: "globalDefault1"},
+						GlobalDefault: true,
+						Value:         90,
+					},
+					{
+						ObjectMeta:    v1.ObjectMeta{Name: "globalDefault2"},
+						GlobalDefault: true,
+						Value:         20,
+					},
+					{
+						ObjectMeta:    v1.ObjectMeta{Name: "globalDefault3"},
+						GlobalDefault: true,
+						Value:         50,
+					},
+				},
+			},
+			expectedPriorityClassName:  "globalDefault2",
+			expectedPriorityClassValue: 20,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.desc, func(t *testing.T) {
+			t.Parallel()
+
+			builder := fake.NewClientBuilder().WithScheme(scheme).WithLists(tt.priorityClassList)
+			client := builder.Build()
+
+			name, value, err := GetPriorityFromPriorityClass(context.Background(), client, tt.priorityClassName)
+			if tt.expectedErr != "" {
+				if err == nil {
+					t.Fatalf("expected error but error is nil")
+				}
+
+				if !strings.Contains(err.Error(), tt.expectedErr) {
+					t.Fatalf("unexpecter error message: got: %s, expected: %s", err.Error(), tt.expectedErr)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			if name != tt.expectedPriorityClassName {
```

Comment:
prefer to use `cmp.Diff` instead. Same to other places.

Reply:
only the error case above since it is a struct, we don't use cmp.Diff when comparing singleton parameters like this.